docs/csharp/tutorials/generate-consume-asynchronous-stream.md

```diff
@@ -0,0 +1,120 @@
+
+## Convert to async streams
+
+Next, convert the `runPagedQueryAsync` to generate an async stream. First, change the signature of `runPagedQueryAsync` to return an `IAsyncEnumerable<JToken>`, and remove the cancellation token and progress objects from the parameter list as shown in the following code:
+
+[!code-csharp[FinishedSignature](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#UpdateSignature)]
+
+The starter code processes each page as the page is retrieved, as shown in the following code:
+
+[!code-csharp[StarterPaging](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#ProcessPage)]
+
+Replace those three lines with the following code:
+
+[!code-csharp[FinishedPaging](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#YieldReturnPage)]
+
+You can also remove the declaration of `finalResults` earlier in this method, and the return statement that follows the loop you modified.
+
+You've finished the changes to generate an async stream. The finished method should resemble the code below:
+
+[!code-csharp[FinishedGenerate](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#GenerateAsyncStream)]
+
+Next, you change the code that consumes the collection to consume the async stream. Find the following code in `Main` that processes the collection of issues:
+
+[!code-csharp[EnumerateOldStyle](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#EnumerateOldStyle)]
+
+Replace all that code with the following `await foreach` loop:
+
+[!code-csharp[FinishedEnumerateAsyncStream](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#EnumerateAsyncStream)]
+
+Run the application again. Contrast its behavior with the behavior of the starter application. The first page of results is enumerated as soon as they are available. There's an observable pause as each new page is requested and retrieved, then the next page's results are quickly enumerated. The `try` / `catch` block is not needed to handle cancellation: the caller could stop enumerating the collection. Progress is clearly reported because the async stream generates results as each page is downloaded.
```

Comment:
```suggestion
Run the application again. Contrast its behavior with the behavior of the starter application. The first page of results is enumerated as soon as they're available. There's an observable pause as each new page is requested and retrieved, then the next page's results are quickly enumerated. The `try` / `catch` block isn't needed to handle cancellation: the caller could stop enumerating the collection. Progress is clearly reported because the async stream generates results as each page is downloaded.
```